glyphs2fontir/src/source.rs

```diff
@@ -90,62 +106,129 @@ impl Source for GlyphsIrSource {
-            .glyphs
-            .get(&glyph_name)
-            .ok_or_else(|| Error::NoStateForGlyph(glyph_name.clone()))?;
-
+    fn create_work_for_one_glyph(&self, glyph_name: &str) -> Result<GlyphIrWork, Error> {
         Ok(GlyphIrWork {
-            glyph_name: glyph_name.clone(),
-            ir_file: self.ir_paths.glyph_ir_file(&glyph_name),
+            glyph_name: glyph_name.to_string(),
         })
     }
 }
 
+struct StaticMetadataWork {
+    font: Arc<Font>,
+}
+
+impl Work for StaticMetadataWork {
+    fn exec(&self, context: &Context) -> Result<(), WorkError> {
+        let font = self.font.as_ref();
+        debug!("Static metadata for {}", font.family_name);
+
+        let mut axis_values = Vec::new();
+        for master in font.font_master.iter() {
+            master
+                .axes_values
+                .as_ref()
+                .ok_or_else(|| {
+                    WorkError::InconsistentAxisDefinitions(format!(
+                        "No axis values for {}",
+                        master.id
+                    ))
+                })?
+                .iter()
+                .enumerate()
+                .for_each(|(idx, value)| {
+                    while axis_values.len() <= idx {
+                        axis_values.push(Vec::new());
+                    }
+                    axis_values[idx].push(value);
+                });
+        }
+
+        let axes = font.axes.as_ref().ok_or(WorkError::NoAxisDefinitions)?;
+        if axes.is_empty() {
+            return Err(WorkError::NoAxisDefinitions);
+        }
+        if axes.len() != axis_values.len() || axis_values.iter().any(|v| v.is_empty()) {
+            return Err(WorkError::InconsistentAxisDefinitions(format!(
+                "Axes {:?} doesn't match axis values {:?}",
+                axes, axis_values
+            )));
+        }
+
+        let default_master_idx = font.default_master_idx();
+
+        let axes = axes
+            .iter()
+            .enumerate()
+            .map(|(idx, a)| {
+                let min = axis_values[idx]
+                    .iter()
+                    .map(|v| OrderedFloat::<f32>(v.into_inner() as f32))
+                    .min()
+                    .unwrap();
+                let max = axis_values[idx]
+                    .iter()
+                    .map(|v| OrderedFloat::<f32>(v.into_inner() as f32))
+                    .max()
+                    .unwrap();
+                let default =
+                    OrderedFloat::<f32>(axis_values[default_master_idx][idx].into_inner() as f32);
+
+                Axis {
+                    name: a.name.clone(),
+                    tag: a.tag.clone(),
+                    hidden: a.hidden.unwrap_or(false),
+                    min,
+                    default,
+                    max,
+                }
+            })
+            .collect();
+
+        let glyph_order = font.glyphs.iter().map(|g| g.glyphname.clone()).collect();
```

Comment:
you may be delighted to learn that Glyphs.app has a `glyphOrder` custom parameter as well (ordered list of glyph names), that works similarly to UFO's public.glyphOrder

https://glyphsapp.com/learn/alter-the-glyph-order-in-font-view

if it's not present it's ok to do like you do, take the implicit ordering of the glyphs in the font as the glyph order.

Reply:
TIL. Added an example of this and a test.

Reply:
When I tried this in the actual Glyphs app the glyphs in the list moved around to match my new glyph order. I have manually adjusted a file to make this not true for test purposes.

Reply:
> When I tried this in the actual Glyphs app the glyphs in the list moved around to match my new glyph order

maybe that's a feature, and we can rely on the font.glyphs order matching any optional 'glyphOrder' custom parameter..
In fact I see that glyphsLib itself takes care of sorting the font.glyphs list by the UFO's glyphOrder when converting UFO => Glyphs

https://github.com/googlefonts/glyphsLib/blob/6763648a91ecadee256d0518c731e35eb311c013/Lib/glyphsLib/builder/builders.py#L555-L560

And when going Glyphs => UFO, glyphsLib always sets UFO public.glyphOrder to the order of the font.glyphs list even when no glyphOrder custom parameter is set.

https://github.com/googlefonts/glyphsLib/blob/6763648a91ecadee256d0518c731e35eb311c013/Lib/glyphsLib/builder/font.py#L88-L91

so it looks consistent with your findings